src/coreclr/src/vm/ilmarshalers.cpp

```diff
@@ -2907,7 +2907,7 @@ MarshalerOverrideStatus ILSafeHandleMarshaler::ArgumentOverride(NDirectStubLinke
             }
 
             // Leave the address of the native handle local as the argument to the native method.
-            pslILDispatch->EmitLDLOCA(dwNativeHandleLocal);
+            EmitLoadNativeLocalAddrForByRefDispatch(pslILDispatch, dwNativeHandleLocal);
```

Comment:
Nit: I am not sure whether factoring this into a helper method helps with readability. `EmitLDLOCA` + `EmitCONV_I` inline may be better. (matter of preference)

Reply:
Not sure that it does either. I started down that path because we missed this in so many places and I was frustrated with the same two lines being in 4 places. My attempt was to make the method name verbose. I am going to leave it as such for now. I am hoping we can avoid touching this again entirely with the source generator approach.